src/api/routes/search.ts

```diff
@@ -41,9 +56,35 @@ export function createSearchRouter(db: DataStore): RouterWithAsync {
     }
     if (hashBuffer !== undefined && hashBuffer.length === 32) {
       const hash = '0x' + hashBuffer.toString('hex');
-      const queryResult = await db.searchHash({ hash });
+      let queryResult: FoundOrNot<DbSearchResult> | FoundOrNot<DbSearchResultWithMetadata> = {
+        found: false,
+      };
+      if (!includeMetadata) {
+        queryResult = await db.searchHash({ hash });
+      } else {
+        queryResult = await searchHashWithMetadata(hash, db);
+      }
       if (queryResult.found) {
-        if (queryResult.result.entity_type === 'block_hash') {
+        if (queryResult.result.entity_type === 'block_hash' && queryResult.result.entity_data) {
+          if (includeMetadata) {
+            const blockData = queryResult.result.entity_data as Block;
+            const blockResult: BlockSearchResult = {
+              found: true,
+              result: {
+                entity_id: queryResult.result.entity_id,
+                entity_type: SearchResultType.BlockHash,
+                block_data: {
+                  canonical: blockData.canonical,
+                  hash: blockData.hash,
+                  parent_block_hash: blockData.parent_block_hash,
+                  burn_block_time: blockData.burn_block_time,
+                  height: blockData.height,
+                },
+                metadata: blockData,
+              },
+            };
+            return blockResult;
```

Comment:
Any reason why the `block` result for `includeMetadata` is different than the others (this one creates a different object rather than assigning `metadata` for the existing one)?

Reply:
wrt my [this comment](https://github.com/hirosystems/stacks-blockchain-api/pull/848#discussion_r760140530) the endpoint `extended/v1/block/{hash}` [responds](https://github.com/hirosystems/stacks-blockchain-api/blob/develop/docs/openapi.yaml#L472-L500) with a `Block` type object with metadata. This is fetched from the [`getBlockWithMetadata`](https://github.com/hirosystems/stacks-blockchain-api/blob/develop/src/datastore/postgres-store.ts#L2554-L2616). However, the search endpoint returns a `DbBlock` object which does not provide information sufficient enough to create enriched block object.